lib/src/default_index/revset_engine.rs

```diff
@@ -461,7 +461,7 @@ where
     }
 }
 
-/// Iterator that intersects two sorted iterators in descending order.
+/// Iterator that intersects two sorted iterators in ascending order.
```

Comment:
Apologies for the dum question, but I could use a bit more explanation on this change.

The order of a union or intersection iterator is entirely determined by the sort order of the _underlying_ iterators, right?  How is it safe to change the comparison function here without changing the sort order of the underlying iterators to match?

Reply:
> The order of a union or intersection iterator is entirely determined by the sort order of the underlying iterators, right? How is it safe to change the comparison function here without changing the sort order of the underlying iterators to match?

The comparison function should be compatible with the underlying iterators as you said. If they differ, these iterators won't work. Are you suggesting that the doc comment is incomplete?

FWIW, the union iterator is basically the same as `Itertools::merge_join_by().map(..)`. I decided to not migrate to itertools as we can't replace the other two anyway.
https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.merge_join_by

Reply:
I guess the iterators are 'descending' both before and after this commit (the second one in the change) with respect to the natural ordering of Positions, but by reversing the comparator and the documentation at the same time everything's consistent (the iterators are _ascending_ now because we changed what _ascending_ and _descending_ mean when changing the the comparison function).  Thank you for rubber-ducking.